consul/src/main/java/com/linecorp/armeria/internal/consul/ConsulClientBuilder.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.consul;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.consul.ConsulConfigSetters;
+
+public class ConsulClientBuilder implements ConsulConfigSetters {
+    public static final String DEFAULT_CONSUL_API_VERSION = "v1";
+    private static final Pattern CONSUL_API_VERSION_PATTERN = Pattern.compile("^v[0-9][-._a-zA-Z0-9]*$");
+
+    private final URI consulUri;
+    private String consulApiVersion = DEFAULT_CONSUL_API_VERSION;
+    @Nullable
+    private String consulToken;
+
+    protected ConsulClientBuilder(URI consulUri) {
```

Comment:
Could be package-local:

```suggestion
    ConsulClientBuilder(URI consulUri) {
```

Reply:
Ok.